src/fu-daemon.c

```diff
@@ -854,24 +701,23 @@ fu_daemon_authorize_install_queue(FuMainAuthHelper *helper_ref)
 	g_autoptr(FuProgress) progress = fu_progress_new(G_STRLOC);
 	gboolean ret;
 
-#ifdef HAVE_POLKIT
 	/* still more things to to authenticate */
 	if (helper->action_ids->len > 0) {
+		FuPolkitAuthorityCheckFlags auth_flags =
+		    FU_POLKIT_AUTHORITY_CHECK_FLAG_ALLOW_USER_INTERACTION;
 		g_autofree gchar *action_id = g_strdup(g_ptr_array_index(helper->action_ids, 0));
-		g_autoptr(PolkitSubject) subject = g_object_ref(helper->subject);
 		g_ptr_array_remove_index(helper->action_ids, 0);
-		polkit_authority_check_authorization(
-		    self->authority,
-		    subject,
-		    action_id,
-		    NULL,
-		    POLKIT_CHECK_AUTHORIZATION_FLAGS_ALLOW_USER_INTERACTION,
-		    NULL,
-		    fu_daemon_authorize_install_cb,
-		    g_steal_pointer(&helper));
+		if (fu_engine_request_has_device_flag(helper->request, FWUPD_DEVICE_FLAG_TRUSTED))
+			auth_flags |= FU_POLKIT_AUTHORITY_CHECK_FLAG_USER_IS_TRUSTED;
+		fu_polkit_authority_check(self->authority,
+					  helper->sender,
```

Comment:
`helper` is NULL if parameters are evaluate right to left (which is often the case). Copying `helper->sender` a temporary before solves segfaults I am getting.